lib/client/Client.js

```diff
@@ -356,6 +356,32 @@ export default class Client {
   }
 
   /**
+   * Destroy multiple {@link Mailbox}es specified to the given _ids_ on the server.<br />
+   * This will issue a {@link Client#setMailboxes} JMAP request under the hoods, passing the correct options.
+   *
+   * @param ids {Array} An array IDs of the mailboxes to destroy. These IDs must be in the right order: Destroy X comes before destroy Y if X is a descendent of Y.
+   *
+   * @return {Promise} A {@link Promise} that eventually resolves to nothing if the {@link Mailbox}es were destroyed successfully.
+   *
+   * @see http://jmap.io/spec.html#destroying-mailboxes
+   */
+  destroyMailboxes(ids) {
+    Utils.assertRequiredParameterIsArrayWithMinimumLength(ids, 'ids', 1);
+
+    return this.setMailboxes({destroy: ids})
+      .then(response => {
+        let failedToDestroyIds = Object.keys(response.notDestroyed);
+
+        if (failedToDestroyIds.length > 0) {
+          // take the first failed one for incrementally debugging
+          let id = failedToDestroyIds[0];
+
+          throw new Error('Failed to destroy ' + id + ', the reason is: ' + (response.notDestroyed[id] || 'missing'));
+        }
+      });
```

Comment:
Such bulk methods do return the whole set response object. It's not up to the library to decide if the call failed when 'some' the of the deletion failed.. 
Maybe there are business rules in the caller that accept some failures as a success, maybe the caller doesn't care, etc. 

Just remove the then block and check success in OP.